pxtblocks/blocklyloader.ts

```diff
@@ -104,14 +104,45 @@ namespace pxt.blocks {
         const value = document.createElement("value");
         value.setAttribute("name", p.definitionName);
 
-        const shadow = document.createElement(isVariable ? "block" : "shadow");
+        const isArray = /(.*)\[\]$/.exec(p.type);
```

Comment:
Also needs to support the alternate syntax for arrays `Array<number>`. See `arrayTypeRegex` at the top of this file